bundles/org.openhab.binding.fineoffsetweatherstation/src/main/java/org/openhab/binding/fineoffsetweatherstation/internal/service/GatewayQueryService.java

```diff
@@ -73,14 +73,20 @@ public GatewayQueryService(FineOffsetGatewayConfiguration config,
 
     protected byte @Nullable [] executeCommand(String command, byte[] request,
             Function<byte[], Boolean> validateResponse) {
-        byte[] buffer = new byte[2028];
-        int bytesRead;
-
         try {
             if (!REQUEST_LOCK.tryLock(30, TimeUnit.SECONDS)) {
-                logger.trace("executeCommand({}): time out while getting lock", command);
+                logger.warn("executeCommand({}): timed out while getting the lock", command);
                 return null;
             }
+        } catch (InterruptedException e) {
+            logger.info("executeCommand({}): was interrupted while getting the lock", command);
```

Comment:
It seems to me that log level could be reduced to **debug** here. For reference: https://www.openhab.org/docs/developer/guidelines.html#f-logging